packages/interaction-kit/src/application.ts

```diff
@@ -135,8 +137,22 @@ export default class Application {
 		return this.#components.get(customID);
 	}
 
-	getCommand(type: ApplicationCommandType, name: string) {
-		return this.#commands.get(type).get(name);
+	getCommand<T extends keyof CommandMap>(
+		type: T,
+		name: string
+	): CommandMapValue<T> | undefined {
+		// I'm not sure why, but this needs to be cast to prevent an error
+		return this.#commands[type].get(name) as CommandMapValue<T> | undefined;
+	}
+
+	getGenericCommand(
```

Comment:
I hate this, but not sure how to do it better (interactions/index.ts needs it in this return format - casting doesn't work).

Reply:
Thanks ali